### PR TITLE
Refactored filtering into a function.

### PR DESCRIPTION
**Pull request by adusca**

### CODE REVIEW

**Comment by armenzg** [comment]
Travis is failling for this.
Running tox will show you the same issues.

Could you also add something similar to what @Gijs did on his other pull request for alltalos.py? (PR#198)

**Comment by adusca** [comment]
alltalos output: https://pastebin.mozilla.org/8833627
triggerbyfilters output: https://pastebin.mozilla.org/8833628

r?

File: mozci/scripts/triggerbyfilters.py [hunk 1]
```diff
@@ -77,22 +72,21 @@ def main():
...
+                     % len(buildernames))
+    if cont.lower() == 'd':
+        LOG.info("The following jobs will be triggered: \n %s" % '\n'.join(buildernames))
+        cont = raw_input("Do you wish to continue? y/n ")
 
-    if len(buildernames) > options.lim:
-        LOG.info('There %i matching buildernames, the limit is %i. If you really want'
-                 'to trigger everything, try again with --limit %i.' % (len(buildernames),
-                                                                        options.lim, options.lim))
+    if cont.lower() != 'y':
```
Comment (armenzg) [inline mozci/scripts/triggerbyfilters.py:89]: If this if/block correct?
It seems that if we answer "yes" we exit.

**Comment by armenzg** [comment]
This looks fantasic!

My only comment is about the if/block.

Could you also update the docs for scripts.rst in a follow up?
Once we release this we should update the docs. Thanks Alice and enjoy your weekend.

**Comment by armenzg** [comment]
Merge when comment about the if/block is checked.

**Pull request merged**